test/cpp/server/credentials_test.cc

```diff
@@ -123,6 +150,71 @@ TEST(
   GPR_ASSERT(server_credentials.get() != nullptr);
 }
 
+TEST_F(
+    CredentialsTest,
+    CoreAPICertificateKeyMatchFailedOnEmptyKey){
+  const char* error_details;
+  grpc_status_code matched = grpc_tls_certificate_key_match(
+      /*private_key=*/"", cert_chain_.c_str(), &error_details);
+  EXPECT_EQ(matched, static_cast<grpc_status_code>(absl::StatusCode::kInvalidArgument));
```

Comment:
Please use GRPC_STATUS_INVALID_ARGUMENT directly here, also for the other tests.